engine/src/main/java/org/hibernate/validator/internal/metadata/aggregated/ValidatableParametersMetaData.java

```diff
@@ -18,14 +25,41 @@
  */
 public class ValidatableParametersMetaData implements Validatable {
 
+	private final List<ParameterMetaData> parameterMetaData;
 	private final Iterable<Cascadable> cascadables;
 
-	public ValidatableParametersMetaData(Iterable<? extends Cascadable> cascadables) {
-		this.cascadables = CollectionHelper.<Cascadable>newHashSet( cascadables );
+	public ValidatableParametersMetaData(List<ParameterMetaData> parameterMetaData) {
+		this.parameterMetaData = Collections.unmodifiableList( parameterMetaData );
+		this.cascadables = Collections.unmodifiableList( parameterMetaData.stream()
+			.filter( p -> p.isCascading() )
+			.collect( Collectors.toList() ) );
```

Comment:
Yes, I think using List would be beneficial in a lot of places. When I looked at the constraints validation ordering, having Sets everywhere annoyed me. I think it would be nice to have the result of a validation as deterministic as possible.